### ISSUE TITLE
Recurring Tasks

### ISSUE THREAD

**Issue by sttrebo** [QTodoTxt/QTodoTxt2#25]
Recurring tasks not working for me right now.  Is this a bug or is this not implemented (yet)?

**Comment by oroulet** [comment]
I never tok the time to understand recurring tasks. but it should not be hard to make it work. PR welcome

**Comment by sttrebo** [comment]
works perfectly for me in V1.x.  kinda depend on it right now.

**Comment by oroulet** [comment]
How is supposed to work? Make a new task when marking it as done? Or every
week/dsy? (But​ this should not work since there is no timer...)

**Comment by sttrebo** [comment]
v1.x makes a new task when marking it as done.  this is perfect for my workflow.  the timer idea is interesting, but i can't see any workflow (for me) that would need this.

**Comment by topas-rec** [comment]
After completing a recurring task, a new task witha new due date is created. The next due date can be configured in the task by defining a <rec> statement. <rec:4d> means the next due date is 4 days in the future from completion date.
<rec:+4d> means the next due date is 4 days in the future from the original due date. When you're lazy then this might be in the past. ;-)

**Comment by oroulet** [comment]
Thanks, now I understand. All your code is still there in qtodotxt2 so we probably just need small changes to make it work

**Comment by topas-rec** [comment]
I use these tasks for things not related to programming.
I use it to remind myself for pc backup, car maintenance and taking out my garbage. It's a real help here.

**Comment by oroulet** [comment]
In fact some code was in a deleted file. so I moved it back refactored a bit and it seems to work now. Let me know

**Comment by sttrebo** [comment]
seems to be working so far, i'll keep testing.

**Comment by sttrebo** [comment]
found not a bug, but would be a great tweak.  due dates get updated just fine.  is it possible to update the threshold date also?  as an example:
(B) Chevron +Personal @Bills due:**2017-06-30 t:2017-06-25** rec:+1m

when i mark this complete i get:
(B) Chevron +Personal @Bills due:**2017-07-30 t:2017-06-25** rec:+1m

not sure how the programming logic would work since you can have the due date update based on original date or the date when it is actually marked as done ( + )

**Comment by topas-rec** [comment]
Oh I missed this whole implementing recurrent tasks in Qtodotxt 1. Qtodotxt 1 could also benefit from this add-on.

**Comment by oroulet** [comment]
The code is exactly the same in both. Just a bit cleaned up in 2

**Comment by sttrebo** [comment]
seems to work for me, thanks

**Comment by topas-rec** [comment]
I just wanted to start making recurring tasks respond to threshold date.

> @oroulet: The code is exactly the same in both. Just a bit cleaned up in 2

I understand: I can just do the changes in QTodoTxt and the code will be available in QTodoTxt2. (That would be ideal!) Is that correct?

**Comment by topas-rec** [comment]
> @sttrebo: as an example:
(B) Chevron +Personal @bills due:2017-06-30 t:2017-06-25 rec:+1m
when i mark this complete i get:
(B) Chevron +Personal @bills due:2017-07-30 t:2017-06-25 rec:+1m

Should it look like this:?
(B) Chevron +Personal @bills due:2017-07-30 t:2017-0**7**-25 rec:+1m

**Comment by sttrebo** [comment]
yes, that looks about right.

**Comment by oroulet** [comment]
@topas-rec i haven't looked at that code for a while, but if I wrote that code is similar, then it probably is ;-) probably better code in qtodotxt2 too

**Comment by topas-rec** [comment]
QTodoTxt2 seems to have threshold date handling. @sttrebo have you tested QTodoTxt2?
I am nearly done with the fix in QTodoTxt.

**Comment by sttrebo** [comment]
yes, been using that feature in QTodoTxt2 for a while and works exactly as it should.
FYI, I use it to track my credit card bills.  Due date when it's due and Threshold 1 week prior to make sure I pay it on time.

Curiosity:  Is there a reason to keep updating QTodoTxt?  I thought QTodoTxt2 was a newer/better version?

**Comment by oroulet** [comment]
If you use qtodotxt2 for accounting then we can say it is stable ;-) there are still some usability issues with qtodotxt2 which I would like to solve before releasing it. Then it is missing packaging code for osx,windowd, etc... I hope somr people will step in and help with that..
